scripts/zones/Apollyon/mobs/Dee_Wapa_the_Desolator.lua

```diff
@@ -2,52 +2,81 @@
-    local X = mob:getXPos();
-    local Y = mob:getYPos();
-    local Z = mob:getZPos();
-    SpawnMob(16933148):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-    SpawnMob(16933147):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-    SpawnMob(16933149):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-    SpawnMob(16933146):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-end;
-
-function onMobFight(mob,target)
-    local mobID = mob:getID();
-    local X = mob:getXPos();
-    local Y = mob:getYPos();
-    local Z = mob:getZPos();
-    local lifepourcent= ((mob:getHP()/mob:getMaxHP())*100);
-    local instancetime = target:getBattlefieldTimeLeft(Central_Temenos_2nd_Floor);
+    mob:setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID())
+end
 
-    if (lifepourcent < 50 and GetNPCByID(16933247):getAnimation() == 8) then
-        SpawnMob(16933151):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-        SpawnMob(16933150):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-        SpawnMob(16933152):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-        GetNPCByID(16933247):setAnimation(9);
+function onMobEngaged(mob, target)
+    local battlefield = mob:getBattlefield()
+    if battlefield:getLocalVar("startTime") == 0 then
+        battlefield:setLocalVar("startTime", battlefield:getRemainingTime())
     end
+    SpawnMob(ID.mob.APOLLYON_CS_MOB[3]+3):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID())
+    SpawnMob(ID.mob.APOLLYON_CS_MOB[3]+4):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID())
+    SpawnMob(ID.mob.APOLLYON_CS_MOB[3]+5):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID())
+    mob:setLocalVar("wave", 1)
+end
 
-    if (instancetime < 13) then
-        if (GetMobByID(16933129):isAlive()) then
-            GetMobByID(16933129):updateEnmity(target);
-        elseif (GetMobByID(16933137):isAlive()) then
-            GetMobByID(16933137):updateEnmity(target);
+function onMobFight(mob, target)
+    local battlefield = mob:getBattlefield()
+    if battlefield then
+        local mobX = mob:getXPos()
+        local mobY = mob:getYPos()
+        local mobZ = mob:getZPos()
+        local remainingTime = battlefield:getRemainingTime()
+        local startTime = battlefield:getLocalVar("startTime")
+        local wave = mob:getLocalVar("wave")
+    
+        if GetMobByID(ID.mob.APOLLYON_CS_MOB[3]+3):isDead() and GetMobByID(ID.mob.APOLLYON_CS_MOB[3]+4):isDead()
+            and GetMobByID(ID.mob.APOLLYON_CS_MOB[3]+5):isDead() and wave == 1
+        then
+            mob:setLocalVar("wave", 2)
+            GetMobByID(ID.mob.APOLLYON_CS_MOB[3]+6):setSpawn(mobX, mobY, mobZ)
+            GetMobByID(ID.mob.APOLLYON_CS_MOB[3]+6):setPos(mobX, mobY, mobZ)
+            SpawnMob(ID.mob.APOLLYON_CS_MOB[3]+6):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID())
+            GetMobByID(ID.mob.APOLLYON_CS_MOB[3]+7):setSpawn(mobX, mobY, mobZ)
+            GetMobByID(ID.mob.APOLLYON_CS_MOB[3]+7):setPos(mobX, mobY, mobZ)
```

Comment:
This can be cleaner with sets, a helper function, and doing this on the onDeath of the summoned mobs (no need for onMobFight)
```lua
-- somewhere.lua
limbus.something.cs_apollyon.yagudo_wave1 = set { ID.mob.APOLLYON_CS_MOB[3]+3, ID.mob.APOLLYON_CS_MOB[3]+4, ID.mob.APOLLYON_CS_MOB[3]+5 }

limbus.something.cs_apollyon.yagudo_wave2 = set { ID.mob.APOLLYON_CS_MOB[3]+6, ID.mob.APOLLYON_CS_MOB[3]+7, ID.mob.APOLLYON_CS_MOB[3]+8 }

tpz.limbus.handleWave = function(mob, race)
    local battlefield = mob:getBattlefield()
    local wave_var = battlefield:getLocalVar(race .. "_wave")
    local wave_set = limbus.something.cs_apollyon[race .. "_wave" .. wave_var]
    if wave_set[mob:getID()] and tpz.limbus.allDead(wave_set) then
        local next_wave = (wave % 2) + 1
        battlefield:setLocalVar(race .. "_wave", next_wave)
        local next_set = limbus.something.cs_apollyon[race .. "_wave" .. next_wave]
        -- get the boss mob based on race somehow; table lookup, if check, whatever - make it race_boss
        local boss_pos = race_boss:getPos()
        for id, _ in pairs(next_set) do
             local mook = GetMobByID(id)
             mook:setSpawn(boss_pos.x, boss_pos.y, boss_pos.z)
             mook:setPos(boss_pos.x, boss_pos.y, boss_pos.z)
             mook:spawn():setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID())
        end
    end
end

-- all yagudo mook luas
function onMobDeath(mob, player, isKiller)
    tpz.limbus.handleWave(mob, "yagudo")
end

-- beastmen mob boss luas
function onMobFight
    -- Nothing! New waves are handled on death of the mooks!
end